File: backend/src/main/java/xyz/engsmyre/moviedescriptiongame/db/MovieStorage.java
package xyz.engsmyre.moviedescriptiongame.db;

import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import java.io.Serial;
import java.io.Serializable;
import java.util.Objects;
import org.springframework.data.annotation.Id;
import org.springframework.data.redis.core.RedisHash;
import org.springframework.data.redis.core.index.Indexed;
import xyz.engsmyre.moviedescriptiongame.tmdb.domain.Movie;

@RedisHash("Movie")
public record MovieStorage(Movie movie,
                           @Id @Indexed String storageKey) implements Serializable {
    @Serial
    private static final long serialVersionUID = 0L;

    public Movie getMovie() {
        return movie;
    }

    public String getStorageKey() {
        return storageKey;
    }

    @Override
    public String storageKey() {
        return storageKey;
    }

    @Override
    public boolean equals(Object obj) {
        if (obj == this) return true;
        if (obj == null || obj.getClass() != this.getClass()) return false;
        var that = (MovieStorage) obj;
        return Objects.equals(this.movie, that.movie) &&
                Objects.equals(this.storageKey, that.storageKey);
    }

    @Override
    public int hashCode() {
        return Objects.hash(movie, storageKey);
    }

    @Override
    public String toString() {
        return "MovieStorage[" +
                "movie=" + movie + ", " +
                "storageKey=" + storageKey + ']';
    }

}

File: backend/src/main/java/xyz/engsmyre/moviedescriptiongame/db/MovieStorage.kt
package xyz.engsmyre.moviedescriptiongame.db

import org.springframework.data.annotation.Id
import org.springframework.data.redis.core.RedisHash
import org.springframework.data.redis.core.index.Indexed
import xyz.engsmyre.moviedescriptiongame.tmdb.domain.Movie
import java.io.Serializable

@RedisHash("Movie")
data class MovieStorage(private val movie : Movie, @Id @Indexed private val storageKey : String) : Serializable {
    fun storageKey() : String {
        return storageKey
    }

    fun getMovie() : Movie {
        return movie
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as MovieStorage

        if (movie != other.movie) return false
        if (storageKey != other.storageKey) return false

        return true
    }

    override fun hashCode(): Int {
        var result = movie.hashCode()
        result = 31 * result + storageKey.hashCode()
        return result
    }

    override fun toString(): String {
        return "MovieStorage(movie=$movie, storageKey='$storageKey')"
    }
}